src/Factory.sol

```diff
@@ -0,0 +1,172 @@
+
+    /// @notice settings for accounts
+    address public settings;
+
+    /// @notice ERC20 token used to interact with markets
+    address public marginAsset;
+
+    /// @notice synthetix address resolver
+    address public addressResolver;
+
+    /// @notice gelato ops
+    address payable public ops;
+
+    /// @notice mapping of account owner to account created
+    mapping(address => address) public ownerToAccount;
+
+    /*//////////////////////////////////////////////////////////////
+                              CONSTRUCTOR
+    //////////////////////////////////////////////////////////////*/
+
+    /// @notice constructor for factory
+    /// @param _owner: owner of factory
+    /// @param _marginAsset: address of ERC20 token used to interact with markets
+    /// @param _addressResolver: address of synthetix address resolver
+    /// @param _settings: address of settings for accounts
+    /// @param _ops: contract address for gelato ops -- must be payable
+    /// @param _implementation: address of account implementation
+    constructor(
+        address _owner,
+        address _marginAsset,
+        address _addressResolver,
+        address _settings,
+        address payable _ops,
+        address _implementation
+    ) Owned(_owner) {
+        marginAsset = _marginAsset;
+        addressResolver = _addressResolver;
+        settings = _settings;
+        ops = _ops;
+        implementation = _implementation;
+    }
+
+    /*//////////////////////////////////////////////////////////////
+                           ACCOUNT DEPLOYMENT
+    //////////////////////////////////////////////////////////////*/
+
+    /// @inheritdoc IFactory
+    function newAccount()
+        external
+        override
+        returns (address payable accountAddress)
+    {
+        /// @dev ensure one account per address
+        if (ownerToAccount[msg.sender] != address(0)) {
+            revert OnlyOneAccountPerAddress(ownerToAccount[msg.sender]);
+        }
+
+        // create account and set beacon to this address (i.e. factory address)
+        accountAddress = payable(address(new AccountProxy(address(this))));
+
+        // update owner to account mapping
+        ownerToAccount[msg.sender] = accountAddress;
+
+        // initialize new account
+        (bool success, bytes memory data) = accountAddress.call(
+            abi.encodeWithSignature(
+                "initialize(address,address,address,address,address,address)",
+                msg.sender, // caller will be set as owner
+                marginAsset,
+                addressResolver,
+                settings,
+                ops,
+                address(this)
+            )
+        );
+        if (!success) revert AccountFailedToInitialize(data);
+
+        // determine version for the following event
+        (success, data) = accountAddress.call(
+            abi.encodeWithSignature("VERSION()")
+        );
+        if (!success) revert AccountFailedToFetchVersion(data);
```

Comment:
What's the case where the version call fails?

Reply:
If the VERSION() function is not defined in the new implementation, it will fail. Also wrote a unit test for this one